### ISSUE TITLE
Gradle 7.4 regression in toolchain support, gradle reports `Invalid encoding name "UTF8"`

### ISSUE THREAD

**Issue by rocketraman** [gradle/gradle#19939]
### Expected Behavior
Toolchain should be initialized correctly.

### Current Behavior
Running Gradle 7.4 with a build that uses toolchains shows this output:

```
[Fatal Error] :1:38: Invalid encoding name "UTF8".
```

Running with `--info`, we see:

```
Selected primary task 'build' from project :
[Fatal Error] :1:38: Invalid encoding name "UTF8".
Java Toolchain auto-detection failed to parse Maven Toolchains located at provider(?)
```

Removing the toolchain configuration from the build eliminates the `Invalid encoding name "UTF8"` error.

The build contains these lines:

```
java {
  toolchain {
    languageVersion.set(JavaLanguageVersion.of(JavaVersion.VERSION_17.majorVersion))
  }
}

kotlin {
  jvmToolchain {
    (this as JavaToolchainSpec).languageVersion.set(JavaLanguageVersion.of(JavaVersion.VERSION_17.majorVersion))
  }
}
```

### Context
Gradle 7.4 regression from 7.3.3.

### Your Environment
JDK 17 (Adoptium)
Linux Fedora 35
Gradle 7.4 via wrapper, same error running standalone

**Comment by ljacomet** [comment]
This is most likely linked to #19046 but should not fail the build. Does it? If yes, please share the full stacktrace by running with `-s`

Can you take a look at the [Maven toolchains](https://maven.apache.org/guides/mini/guide-using-toolchains.html) file on your system? I wonder if it declares its encoding as `UTF8` instead of `UTF-8`

**Comment by rocketraman** [comment]
> This is most likely linked to #19046 but should not fail the build.

It does not fail the build.

> Can you take a look at the [Maven toolchains](https://maven.apache.org/guides/mini/guide-using-toolchains.html) file on your system? I wonder if it declares its encoding as `UTF8` instead of `UTF-8`

That was it, thank you.

**Comment by rocketraman** [comment]
Leaving this open just in case you think the error handling for this case should be improved. If not, I think it makes sense to close it.

**Comment by ljacomet** [comment]
Need to improve the error message, that `provider(?)` at the end should be the toolchains file location.
Also need to check if we can prevent the `[Fatal Error]` to be printed at all, since this is not fatal in our context.

**Comment by Marcono1234** [comment]
It is likely that more users will run into this because the Maven toolchains guide contained this incorrect encoding name in the past, see https://github.com/apache/maven-site/pull/252.

**Comment by big-guy** [comment]
@rocketraman We were trying to add a test to check this behavior and haven't been able to reproduce it exactly. Do you know if you have anything else special in your environment?  e.g., what version of Java were you using to run Gradle?  What was your locale and file encoding set to?

**Comment by rocketraman** [comment]
Nothing special. Java 17 (as per the OP). Locale and file encoding should all be UTF-8.

Did you try to replicate with "UTF8" set as the encoding in `.m2/toolchain.xml`? That's all I need to do to repro it.

**Comment by big-guy** [comment]
Yes, I was able to reproduce it by having a file with an encoding that's gibberish (`asdf`), but not `UTF8`.  We tried running this on Java 8, 11 and 17. 

Were there any other plugins applied to the build script?

Could you run again with the `UTF8` encoding and `--debug` and send us the stacktrace? 

Could you post the toolchains file too?

**Comment by rocketraman** [comment]
Here is a repo that successfully reproduces it for me: https://github.com/rocketraman/test-gradle-19939.

Here is the debug log and toolchains.xml:

[build-debug-logs.txt](https://github.com/gradle/gradle/files/8584686/build-debug-logs.txt)
[toolchains.zip](https://github.com/gradle/gradle/files/8584691/toolchains.zip)

**Comment by rocketraman** [comment]
Relevant section of the debug log:

```
2022-04-28T12:41:34.184-0400 [DEBUG] [org.gradle.jvm.toolchain.internal.MavenToolchainsInstallationSupplier] Java Toolchain auto-detection failed to parse Maven Toolchains located at provider(?)
org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 38; Invalid encoding name "UTF8".
  at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
  at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
  at java.xml/javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:122)
  at org.gradle.jvm.toolchain.internal.MavenToolchainsInstallationSupplier.findCandidates(MavenToolchainsInstallationSupplier.java:72)
  at org.gradle.jvm.toolchain.internal.AutoDetectingInstallationSupplier.get(AutoDetectingInstallationSupplier.java:41)
  at org.gradle.jvm.toolchain.internal.AutoDetectingInstallationSupplier.get(AutoDetectingInstallationSupplier.java:26)
  at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
  at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
  at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
  at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
  at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:960)
  at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:934)
  at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
  at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165
```

**Comment by jvandort** [comment]
@rocketraman Thanks for sending over a reproducer. I was able to replicate this issue on my side and after some further investigation, it seems this specific bug will be fixed in Gradle 7.5. We were not able to reproduce it without using a standalone build since our tests run against the yet-to-be-released builds of Gradle, where this bug was already fixed. 

Closing as we have already merged code to improve the logging and have proven that this bug is fixed. See the linked commit below which removes the xerces lib, which was producing this error. 

https://github.com/gradle/gradle/commit/c334351889661a8d8d4d390c0ea17d92318d3a09
